semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -2244,11 +2241,839 @@ occurrences. We intend to improve on this in the future.
+  </tr>
+</table>
+
+Notes:
+* Class members must appear in the order specified below. This requirement is
+  necessary to compute consistent method symbol disambiguators in the Scala
+  compiler where static and non-static members have separate owners.
+  * non-static members first, following the same order as they appear in the
+    original source.
+  * static members secondly, following the same order as they appear in the
+    original source
+* A Java class maps to a single symbol with type `ClassInfoType` including all
+  static and non-static members. This departs from the Scala compiler internal
+  representation of Java classes where non-static members are grouped under a
+  `CLASS` symbol and static members are grouped under an `OBJECT` symbol.
+* Supported properties for `CLASS` symbols are
+  * `FINAL` set for all final classes
+  * `ABSTRACT` set for all abstract classes
+  * `STATIC` set for static inner classes
+  * `ENUM` set for enum types
+* Class declarations support [all Java accessibilities](#java-accessibility).
+* Class members without explicit access modifiers have accessibility
+  `PRIVATE_WITHIN` within the enclosing package.
+
+**Enum declarations** [\[84\]][84] are represented by a single symbol
+with the `CLASS` kind.
+
+```java
+package a;
+public enum Coin {
+  PENNY, NICKEL
+}
+```
+
+<table>
+  <tr>
+    <td><b>Definition</b></td>
+    <td width="200px"><b>Symbol</b></td>
+    <td><b>Kind</b></td>
+    <td><b>Signature</b></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(), List(&lt;Enum&lt;Coin&gt;&gt;), List(&lt;PENNY&gt;, &lt;NICKEL&gt;))</code></td>
+  </tr>
+  <tr>
+    <td><code>PENNY</code></td>
+    <td><code>a.Coin#PENNY.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;, List())</code></td>
+  </tr>
+  <tr>
+    <td><code>NICKEL</code></td>
+    <td><code>a.Coin#NICKEL.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;, List())</code></td>
+  </tr>
+  <tr>
+    <td></td>
+    <td><code>a.Coin#values().</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Array&gt;, List(&lt;Coin&gt;)))</code></td>
+  </tr>
+  <tr>
+    <td></td>
+    <td><code>a.Coin#valueOf(String).</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Coin&gt;, List()))</code></td>
+  </tr>
+</table>
+
+Notes:
+* Enum declarations follow the same rules as class declarations.
+* Supported properties for enum declarations are:
+  * `FINAL`: implicitly set for all enum declarations.
+  * `STATIC`: implicitly set for all enum declarations.
+  * `ENUM`: implicitly set for all enum declarations.
+* JLS mandates the following synthetic members for enum declarations
+  [\[86\]][86]:
+  * Enum fields have kind `FIELD`, properties `FINAL`, `STATIC` and `ENUM`,
```

Comment:
Should this bullet appear in the top-level above "JLS mandates"?

Reply:
Enum fields are also synthetic, so I thought it would be good to group both them and the synthetic methods.